configs/components/facter.rb

```diff
@@ -110,7 +110,7 @@
     skip_blkid = 'OFF'
   elsif platform.is_rpm?
     if (platform.is_el? && platform.os_version.to_i >= 6) || (platform.is_sles? && platform.os_version.to_i >= 11) || platform.is_fedora?
-      pkg.build_requires "libblkid-devel" unless platform.architecture == "s390x"
+      pkg.build_requires "libblkid-devel" unless platform.is_cross_compiled?
```

Comment:
I'll update this with a comment - the reason this is necessary is if we have libblkid-devel installed, facter will try to link against the native (x86_64) library and not the one in our sysroot. So we want to avoid having this package installed in all cross-compiled situations.